ipatests/test_integration/test_adtrust_install.py

```diff
@@ -314,3 +314,376 @@ def test_ipa_user_s4u2self_pac(self):
+        value
+        """
+        rid_base_value = "103.2"
+        msg = (
+            "ipa-adtrust-install: error: option " "--rid-base: "
+            "invalid integer value: '{}'"
+        ).format(rid_base_value)
+        result = self.master.run_command(
+            [
+                "ipa-adtrust-install",
+                "-A",
+                "admin",
+                "-a",
+                self.master.config.admin_password,
+                "--rid-base",
+                rid_base_value,
+                "-U",
+            ],
+            raiseonerr=False,
+        )
+        assert msg in result.stderr_text
+
+    def test_adtrust_install_with_invalid_secondary_rid_base(self):
+        """
+        Test to check adtrust install with invalid secondary rid-base
+        value
+        """
+        sec_rid_base_value = "103.2"
+        msg = (
+            "ipa-adtrust-install: error: option "
+            "--secondary-rid-base: invalid integer value: '{}'"
+        ).format(sec_rid_base_value)
+        result = self.master.run_command(
+            [
+                "ipa-adtrust-install",
+                "-A",
+                "admin",
+                "-a",
+                self.master.config.admin_password,
+                "--secondary-rid-base",
+                sec_rid_base_value,
+                "-U",
+            ],
+            raiseonerr=False,
+        )
+        assert msg in result.stderr_text
+
+    def test_adtrust_reinstall_updates_ipaNTFlatName_attribute(self):
+        """
+        Test checks that reinstalling ipa-adtrust-install with
+        new netbios name reflects changes in ipaNTFlatName attribute
+        and ipa trustconfig-show also reflects the same.
+        """
+        cmd_args = ["ldapsearch", "-Y", "GSSAPI", "(ipaNTFlatName=*)"]
+        netbios_name = "TEST8REALM"
+        cmd = self.master.run_command(
+            [
+                "ipa-adtrust-install",
+                "-a",
+                self.master.config.admin_password,
+                "--netbios-name",
+                netbios_name,
+                "-U",
+            ]
+        )
+        assert cmd.returncode == 0
+        cmd = self.master.run_command(cmd_args)
+        cmd1 = self.master.run_command(["ipa", "trustconfig-show"])
+        assert "ipaNTFlatName: {}".format(netbios_name) in \
+               cmd.stdout_text and cmd1.stdout_text
+
+    def test_adtrust_install_without_ipa_installed(self):
+        """
+        Tests checks that ipa-adrust-install warns when
+        ipa is not installed on the system
+        """
+        msg = (
+            "IPA is not configured on this system."
+        )
+        tasks.uninstall_master(self.master)
+        result = self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password, "-U"], raiseonerr=False)
+        assert msg in result.stderr_text
+
+    def test_samba_credential_cache_is_removed_post_uninstall(self):
```

Comment:
test_samba_credential_cache_is_removed_post_uninstall is moved ahead of test_adtrust_install_without_ipa_installed and both the tests have been  moved to the end, so that they are executed atlast.